sig/datadog/profiling/stack_recorder.rbs

```diff
@@ -22,9 +22,16 @@ module Datadog
         bool timeline_enabled,
       ) -> true
 
-      def serialize: () -> untyped
+      type serialized_profile_data = [
+        ::Time, # Start
+        ::Time, # Finish
+        String, # Serialized PProf
+        ::Hash[::Symbol, untyped], # Hash
+      ]
 
-      def self._native_serialize: (Datadog::Profiling::StackRecorder recorder_instance) -> [:ok | :error, untyped]
+      def serialize: () -> (serialized_profile_data | nil)
+
+      def self._native_serialize: (Datadog::Profiling::StackRecorder recorder_instance) -> ([:ok, serialized_profile_data] | [:error, ::String])
```

Comment:
Minor: This is not quite correct, the signature of `_native_serialize` is only (Time, Time, String)`, e.g. it does not include the stats hash.

Reply:
But it does? https://github.com/DataDog/dd-trace-rb/blob/0221fbebc9849f7a2b30984b1670eea1eeaf8ad6/ext/datadog_profiling_native_extension/stack_recorder.c?plain=1#L571

Reply:
Aaah you're completely right, I missed it